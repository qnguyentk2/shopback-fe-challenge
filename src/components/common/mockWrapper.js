import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import branches from '../../reducers/branches';

const mockStore = configureStore([thunk]);

export default function getWrapper(
  component,
  useMemory = false,
  useShallow = false,
) {
  const store = mockStore({
    branches,
  });
  if (useShallow) {
    return shallow(<Provider store={store}>{component}</Provider>);
  }
  if (!useMemory) {
    return mount(<Provider store={store}>{component}</Provider>);
  }
  return mount(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>,
  );
}
