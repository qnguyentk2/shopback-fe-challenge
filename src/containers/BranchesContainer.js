import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Branches from "../components/Branches";
import * as branchesAction from "../actions/branches";

class BranchesContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isMobile: false,
    };
  }

  componentDidMount() {
    this.props.fetchBranchesData();
    window.addEventListener("resize", this.handleResize);
    this.handleResize();
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.handleResize);
  }

  handleResize = () => {
    let currentHideNav = window.innerWidth <= 760;
    if (currentHideNav !== this.state.hideNav) {
      this.setState({ isMobile: currentHideNav });
    }
  };

  render() {
    const { categories, stores } = this.props;
    const { isMobile } = this.state;

    return (
      <Branches categories={categories} stores={stores} isMobile={isMobile} />
    );
  }
}

const mapStateToProps = (state) => {
  const { categories, stores } = state.branches;

  return { categories, stores };
};

const mapDispatchToProps = {
  fetchBranchesData: branchesAction.actFetchBranchesData
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BranchesContainer);

BranchesContainer.propTypes = {
  categories: PropTypes.objectOf(PropTypes.any),
  stores: PropTypes.objectOf(PropTypes.any),
  fetchBranchesData: PropTypes.func,
};

BranchesContainer.defaultProps = {
  categories: {},
  stores: {},
  fetchBranchesData: () => {},
};

