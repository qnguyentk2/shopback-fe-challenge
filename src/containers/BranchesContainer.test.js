import React from 'react';
import mockWrapper from '../components/common/mockWrapper';
// import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import '../setupTests';

import BranchesContainer from './BranchesContainer';

test('renders without crashing', () => {
  const wrapper = mockWrapper(<BranchesContainer />, true);
  expect(toJSON(wrapper)).toMatchSnapshot();
  // expect(wrapper.html()).toMatch(/Popular/);
});