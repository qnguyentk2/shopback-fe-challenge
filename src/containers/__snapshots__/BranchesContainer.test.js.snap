// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "tfbgvt",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "tfbgvt",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(BranchesContainer)>
        <BranchesContainer
          categories={Object {}}
          fetchBranchesData={[Function]}
          stores={Object {}}
        >
          <Branches
            categories={Object {}}
            isMobile={false}
            stores={Object {}}
          >
            <div
              className="branches"
            >
              <div
                className="wrapper"
              >
                <div
                  className="branches__tabs"
                >
                  <Tabs
                    activeKey="0"
                    hideAdd={false}
                    tabPosition="top"
                  >
                    <Tabs
                      activeKey="0"
                      className="ant-tabs-line"
                      destroyInactiveTabPane={false}
                      direction="ltr"
                      hideAdd={false}
                      keyboard={true}
                      navWrapper={[Function]}
                      onChange={[Function]}
                      prefixCls="ant-tabs"
                      renderTabBar={[Function]}
                      renderTabContent={[Function]}
                      style={Object {}}
                      tabBarPosition="top"
                      tabPosition="top"
                    >
                      <div
                        className="ant-tabs ant-tabs-top ant-tabs-line"
                        onScroll={[Function]}
                        style={Object {}}
                      >
                        <TabBar
                          activeKey="0"
                          animated={true}
                          direction="ltr"
                          hideAdd={false}
                          key="tabBar"
                          navWrapper={[Function]}
                          onKeyDown={[Function]}
                          onTabClick={[Function]}
                          panels=""
                          prefixCls="ant-tabs"
                          tabBarExtraContent={null}
                          tabBarPosition="top"
                          tabPosition="top"
                          type="line"
                        >
                          <ScrollableInkTabBar
                            activeKey="0"
                            animated={true}
                            className="ant-tabs-top-bar"
                            direction="ltr"
                            extraContent={null}
                            hideAdd={false}
                            inkBarAnimated={true}
                            navWrapper={[Function]}
                            nextIcon={
                              <span
                                className="ant-tabs-tab-next-icon"
                              >
                                <ForwardRef(RightOutlined)
                                  className="ant-tabs-tab-next-icon-target"
                                />
                              </span>
                            }
                            onKeyDown={[Function]}
                            onTabClick={[Function]}
                            panels=""
                            prefixCls="ant-tabs"
                            prevIcon={
                              <span
                                className="ant-tabs-tab-prev-icon"
                              >
                                <ForwardRef(LeftOutlined)
                                  className="ant-tabs-tab-prev-icon-target"
                                />
                              </span>
                            }
                            tabBarExtraContent={null}
                            tabBarPosition="top"
                            tabPosition="top"
                            type="line"
                          >
                            <SaveRef>
                              <TabBarRootNode
                                activeKey="0"
                                animated={true}
                                className="ant-tabs-top-bar"
                                direction="ltr"
                                extraContent={null}
                                getRef={[Function]}
                                hideAdd={false}
                                inkBarAnimated={true}
                                navWrapper={[Function]}
                                nextIcon={
                                  <span
                                    className="ant-tabs-tab-next-icon"
                                  >
                                    <ForwardRef(RightOutlined)
                                      className="ant-tabs-tab-next-icon-target"
                                    />
                                  </span>
                                }
                                onKeyDown={[Function]}
                                onTabClick={[Function]}
                                panels=""
                                prefixCls="ant-tabs"
                                prevIcon={
                                  <span
                                    className="ant-tabs-tab-prev-icon"
                                  >
                                    <ForwardRef(LeftOutlined)
                                      className="ant-tabs-tab-prev-icon-target"
                                    />
                                  </span>
                                }
                                saveRef={[Function]}
                                style={Object {}}
                                tabBarExtraContent={null}
                                tabBarPosition="top"
                                tabPosition="top"
                                type="line"
                              >
                                <div
                                  className="ant-tabs-bar ant-tabs-top-bar"
                                  onKeyDown={[Function]}
                                  role="tablist"
                                  style={Object {}}
                                  tabIndex={-1}
                                >
                                  <ScrollableTabBarNode
                                    activeKey="0"
                                    animated={true}
                                    className="ant-tabs-top-bar"
                                    direction="ltr"
                                    extraContent={null}
                                    getRef={[Function]}
                                    hideAdd={false}
                                    inkBarAnimated={true}
                                    navWrapper={[Function]}
                                    nextIcon={
                                      <span
                                        className="ant-tabs-tab-next-icon"
                                      >
                                        <ForwardRef(RightOutlined)
                                          className="ant-tabs-tab-next-icon-target"
                                        />
                                      </span>
                                    }
                                    onKeyDown={[Function]}
                                    onNextClick={[Function]}
                                    onPrevClick={[Function]}
                                    onTabClick={[Function]}
                                    panels=""
                                    prefixCls="ant-tabs"
                                    prevIcon={
                                      <span
                                        className="ant-tabs-tab-prev-icon"
                                      >
                                        <ForwardRef(LeftOutlined)
                                          className="ant-tabs-tab-prev-icon-target"
                                        />
                                      </span>
                                    }
                                    saveRef={[Function]}
                                    scrollAnimated={true}
                                    tabBarExtraContent={null}
                                    tabBarPosition="top"
                                    tabPosition="top"
                                    type="line"
                                  >
                                    <div
                                      className="ant-tabs-nav-container"
                                      key="container"
                                    >
                                      <span
                                        className="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
                                        onClick={null}
                                        onTransitionEnd={[Function]}
                                        unselectable="unselectable"
                                      >
                                        <span
                                          className="ant-tabs-tab-prev-icon"
                                        >
                                          <ForwardRef(LeftOutlined)
                                            className="ant-tabs-tab-prev-icon-target"
                                          >
                                            <AntdIcon
                                              className="ant-tabs-tab-prev-icon-target"
                                              icon={
                                                Object {
                                                  "icon": Object {
                                                    "attrs": Object {
                                                      "focusable": "false",
                                                      "viewBox": "64 64 896 896",
                                                    },
                                                    "children": Array [
                                                      Object {
                                                        "attrs": Object {
                                                          "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                                        },
                                                        "tag": "path",
                                                      },
                                                    ],
                                                    "tag": "svg",
                                                  },
                                                  "name": "left",
                                                  "theme": "outlined",
                                                }
                                              }
                                            >
                                              <span
                                                aria-label="left"
                                                className="anticon anticon-left ant-tabs-tab-prev-icon-target"
                                                role="img"
                                              >
                                                <IconReact
                                                  className=""
                                                  icon={
                                                    Object {
                                                      "icon": Object {
                                                        "attrs": Object {
                                                          "focusable": "false",
                                                          "viewBox": "64 64 896 896",
                                                        },
                                                        "children": Array [
                                                          Object {
                                                            "attrs": Object {
                                                              "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                                            },
                                                            "tag": "path",
                                                          },
                                                        ],
                                                        "tag": "svg",
                                                      },
                                                      "name": "left",
                                                      "theme": "outlined",
                                                    }
                                                  }
                                                >
                                                  <svg
                                                    aria-hidden="true"
                                                    className=""
                                                    data-icon="left"
                                                    fill="currentColor"
                                                    focusable="false"
                                                    height="1em"
                                                    key="svg-left"
                                                    viewBox="64 64 896 896"
                                                    width="1em"
                                                  >
                                                    <path
                                                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                                                      key="svg-left-svg-0"
                                                    />
                                                  </svg>
                                                </IconReact>
                                              </span>
                                            </AntdIcon>
                                          </ForwardRef(LeftOutlined)>
                                        </span>
                                      </span>
                                      <span
                                        className="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
                                        onClick={null}
                                        unselectable="unselectable"
                                      >
                                        <span
                                          className="ant-tabs-tab-next-icon"
                                        >
                                          <ForwardRef(RightOutlined)
                                            className="ant-tabs-tab-next-icon-target"
                                          >
                                            <AntdIcon
                                              className="ant-tabs-tab-next-icon-target"
                                              icon={
                                                Object {
                                                  "icon": Object {
                                                    "attrs": Object {
                                                      "focusable": "false",
                                                      "viewBox": "64 64 896 896",
                                                    },
                                                    "children": Array [
                                                      Object {
                                                        "attrs": Object {
                                                          "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                                        },
                                                        "tag": "path",
                                                      },
                                                    ],
                                                    "tag": "svg",
                                                  },
                                                  "name": "right",
                                                  "theme": "outlined",
                                                }
                                              }
                                            >
                                              <span
                                                aria-label="right"
                                                className="anticon anticon-right ant-tabs-tab-next-icon-target"
                                                role="img"
                                              >
                                                <IconReact
                                                  className=""
                                                  icon={
                                                    Object {
                                                      "icon": Object {
                                                        "attrs": Object {
                                                          "focusable": "false",
                                                          "viewBox": "64 64 896 896",
                                                        },
                                                        "children": Array [
                                                          Object {
                                                            "attrs": Object {
                                                              "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                                            },
                                                            "tag": "path",
                                                          },
                                                        ],
                                                        "tag": "svg",
                                                      },
                                                      "name": "right",
                                                      "theme": "outlined",
                                                    }
                                                  }
                                                >
                                                  <svg
                                                    aria-hidden="true"
                                                    className=""
                                                    data-icon="right"
                                                    fill="currentColor"
                                                    focusable="false"
                                                    height="1em"
                                                    key="svg-right"
                                                    viewBox="64 64 896 896"
                                                    width="1em"
                                                  >
                                                    <path
                                                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                                                      key="svg-right-svg-0"
                                                    />
                                                  </svg>
                                                </IconReact>
                                              </span>
                                            </AntdIcon>
                                          </ForwardRef(RightOutlined)>
                                        </span>
                                      </span>
                                      <div
                                        className="ant-tabs-nav-wrap"
                                      >
                                        <div
                                          className="ant-tabs-nav-scroll"
                                        >
                                          <div
                                            className="ant-tabs-nav ant-tabs-nav-animated"
                                          >
                                            <TabBarTabsNode
                                              activeKey="0"
                                              animated={true}
                                              className="ant-tabs-top-bar"
                                              direction="ltr"
                                              extraContent={null}
                                              hideAdd={false}
                                              inkBarAnimated={true}
                                              navWrapper={[Function]}
                                              nextIcon={
                                                <span
                                                  className="ant-tabs-tab-next-icon"
                                                >
                                                  <ForwardRef(RightOutlined)
                                                    className="ant-tabs-tab-next-icon-target"
                                                  />
                                                </span>
                                              }
                                              onKeyDown={[Function]}
                                              onTabClick={[Function]}
                                              panels=""
                                              prefixCls="ant-tabs"
                                              prevIcon={
                                                <span
                                                  className="ant-tabs-tab-prev-icon"
                                                >
                                                  <ForwardRef(LeftOutlined)
                                                    className="ant-tabs-tab-prev-icon-target"
                                                  />
                                                </span>
                                              }
                                              renderTabBarNode={null}
                                              saveRef={[Function]}
                                              tabBarExtraContent={null}
                                              tabBarGutter={null}
                                              tabBarPosition="top"
                                              tabPosition="top"
                                              type="line"
                                            >
                                              <div />
                                            </TabBarTabsNode>
                                            <InkTabBarNode
                                              activeKey="0"
                                              animated={true}
                                              className="ant-tabs-top-bar"
                                              direction="ltr"
                                              extraContent={null}
                                              getRef={[Function]}
                                              hideAdd={false}
                                              inkBarAnimated={true}
                                              navWrapper={[Function]}
                                              nextIcon={
                                                <span
                                                  className="ant-tabs-tab-next-icon"
                                                >
                                                  <ForwardRef(RightOutlined)
                                                    className="ant-tabs-tab-next-icon-target"
                                                  />
                                                </span>
                                              }
                                              onKeyDown={[Function]}
                                              onTabClick={[Function]}
                                              panels=""
                                              prefixCls="ant-tabs"
                                              prevIcon={
                                                <span
                                                  className="ant-tabs-tab-prev-icon"
                                                >
                                                  <ForwardRef(LeftOutlined)
                                                    className="ant-tabs-tab-prev-icon-target"
                                                  />
                                                </span>
                                              }
                                              saveRef={[Function]}
                                              styles={Object {}}
                                              tabBarExtraContent={null}
                                              tabBarPosition="top"
                                              tabPosition="top"
                                              type="line"
                                            >
                                              <div
                                                className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                                                key="inkBar"
                                              />
                                            </InkTabBarNode>
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                  </ScrollableTabBarNode>
                                </div>
                              </TabBarRootNode>
                            </SaveRef>
                          </ScrollableInkTabBar>
                        </TabBar>
                        <TabContent
                          activeKey="0"
                          animated={true}
                          animatedWithMargin={true}
                          className="ant-tabs-top-content"
                          destroyInactiveTabPane={false}
                          direction="ltr"
                          key="tabContent"
                          onChange={[Function]}
                          prefixCls="ant-tabs"
                          tabBarPosition="top"
                        >
                          <div
                            className="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </TabContent>
                      </div>
                    </Tabs>
                  </Tabs>
                </div>
              </div>
            </div>
          </Branches>
        </BranchesContainer>
      </Connect(BranchesContainer)>
    </Router>
  </MemoryRouter>
</Provider>
`;
